doc/user-manual/language/cubical.lagda.rst

```diff
@@ -97,10 +113,576 @@ of the interval ``i0`` and ``i1`` are the bottom and top elements, respectively
     p₇ = reflI
     p₈ = reflI
 
+
+The core idea of Homotopy Type Theory and Univalent Foundations is a
+correspondence between paths (as in topology) and (proof-relevant)
+equality (as in Martin-Löf's identity type). This correspondence is
+taken very literally in Cubical Agda where a path in a type ``A`` is
+represented like a function out of the interval, ``I → A``. In fact a
+path type is in fact a special case of the more general built-in
+heterogeneous path types:
+
+.. code-block:: agda
+
+   PathP : ∀ {ℓ} (A : I → Set ℓ) → A i0 → A i1 → Set ℓ
+
+   -- Non dependent path types
+   Path : ∀ {ℓ} (A : Set ℓ) → A → A → Set ℓ
+   Path A a b = PathP (λ _ → A) a b
+
+The central notion of equality in Cubical Agda is hence heterogeneous
+(which is sometimes called "John Major equality"). To define paths we
+use λ-abstractions and to apply them we use regular application.  For
```

Comment:
AFAIU John Major corresponds strictly to Heterogeneous equality + K. The pun being that
we pretend everyone is equal no matter the type but in the end only elements of the same
type are actually treated equally (being substituted). Same as the "classless society" which,
surprise, surprise, happens to still have classes and inequalities.

I think cubical is a different approach: you *expect* a proof that the two endpoints' types are
equal before even talking about them being equal.

Anyway, I think John Major will only add confusion in this context.

Reply:
More than UIP, the issue is where the equality between types resides. John Major is more like this: 
```agda
JM : {A B : Set} -> A -> B -> Set _
JM {A} {B} x y = Σ (p : A ≡ B) \ p → PathP (\ i → p i) x y
```
Which indeed is hard to use for anything without K, because that `p` could be any equivalence.

We could reference the `PathOver` notion instead?

Reply:
Thanks for clarifying what JM equality is. I was indeed mixing it up with `PathOver`.